src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs

```diff
@@ -538,56 +519,48 @@ public async Task<Either<ActionResult, List<ReleaseSeriesTableEntryViewModel>>>
                         if (seriesItem.ReleaseId != null && (
                                 seriesItem.LegacyLinkDescription != null || seriesItem.LegacyLinkUrl != null))
                         {
-                            throw new ArgumentException(
-                                $"LegacyLink details shouldn't be set if ReleaseId is set. ReleaseSeriesItem: {seriesItem.Id}");
+                            throw new ArgumentException("LegacyLink details shouldn't be set if ReleaseId is set.");
                         }
 
                         if (seriesItem.ReleaseId == null && (
                                 seriesItem.LegacyLinkDescription == null || seriesItem.LegacyLinkUrl == null))
                         {
-                            throw new ArgumentException(
-                                $"LegacyLink details should be set if ReleaseId is null. ReleaseSeriesItem: {seriesItem.Id}");
+                            throw new ArgumentException("LegacyLink details should be set if ReleaseId is null.");
                         }
                     }
 
                     // Check all publication releases are included in updatedReleaseSeriesItems
-                    var publicationReleaseIds = publication.ReleaseVersions
-                        .Select(rv => rv.ReleaseId)
-                        .Distinct()
-                        .ToList();
+                    var publicationReleaseIds = await _context.Releases
+                        .Where(r => r.PublicationId == publicationId)
+                        .Select(r => r.Id)
+                        .ToListAsync();
```

Comment:
I think this means rather than one query with a join, we now do two separate queries? I'm not fussed either way, but interested why you decided to make the change

Reply:
I'm not sure I follow where the query before with the join was or where the two separate queries come from.

The code in the comment here should be generating something equivalent to `SELECT Id fFROM Releases WHERE PublicationId = <publicationId>`

Reply:
Ah I see your point now, yes you are correct, there was a join between `Publications` and `ReleaseVersions` before but including the `ReleaseVersions` collection in that way that meant every column of `ReleaseVersions` would be returned for every publication in the result of the query. This was a lot more data about release versions than what was needed which is just the release id.

Now we only select the id's in the second query. We've also changed this to get the Id's from `Releases` directly by publication id without `ReleaseVersions` meaning there's no longer any need for a distinct.